ouroboros-consensus-test/src/Test/Util/Serialisation/Golden.hs

```diff
@@ -172,19 +174,24 @@ goldenTestCBOR testName example enc goldenFile =
               ]
 
 goldenTests ::
-     HasCallStack
+     forall a. HasCallStack
   => TestName
   -> Labelled a
   -> (a -> Encoding)
   -> FilePath  -- ^ Folder containing the golden files
   -> TestTree
-goldenTests testName examples enc goldenFolder
-  | nub labels /= labels
-  = error $ "Examples with the same label for " <> testName
+goldenTests testName examplesFull enc goldenFolder
+  | not (null duplicateLabels)
+  = error $ "Examples with the same label for " <> testName <> ". Duplicate labels:\n" <> show duplicateLabels
   | [(Nothing, example)] <- examples
     -- If there's just a single unlabelled example, no need for grouping,
     -- which makes the output more verbose.
   = goldenTestCBOR testName example enc (goldenFolder </> testName)
+  | not (null (drop maxSize examplesFull))
```

Comment:
If this guard is the first, then you don't need to distinguish between `examples` and `examplesFull`.

Reply:
Good idea!